### PR TITLE
Fix layout.html being loaded instead of Cypress

### LINKED ISSUE #2891: Cypress loads wrong page on start
### Current behavior:

When I try to run cypress tests (the default examples that come after installing cypress) it starts the app properly, but the tests load the wrong page.
It doesn't load the Cypress test page inside the browser, it loads a html file I have in project.

The page that the browser opens is saved in `<project_dir>/views/layout.html`
whenever I run `npx cypress open` inside the `project_dir`.

The project directory looks like this:
```
→ tree
.
├── cypress
│   ├── fixtures
│   │   └── example.json
│   ├── integration
│   │   └── examples
│   │       ├── actions.spec.js
│   │       ├── aliasing.spec.js
│   │       ├── assertions.spec.js
│   │       ├── connectors.spec.js
│   │       ├── cookies.spec.js
│   │       ├── cypress_api.spec.js
│   │       ├── files.spec.js
│   │       ├── local_storage.spec.js
│   │       ├── location.spec.js
│   │       ├── misc.spec.js
│   │       ├── navigation.spec.js
│   │       ├── network_requests.spec.js
│   │       ├── querying.spec.js
│   │       ├── spies_stubs_clocks.spec.js
│   │       ├── traversal.spec.js
│   │       ├── utilities.spec.js
│   │       ├── viewport.spec.js
│   │       ├── waiting.spec.js
│   │       └── window.spec.js
│   ├── plugins
│   │   └── index.js
│   └── support
│       ├── commands.js
│       └── index.js
├── cypress.json
└── views
    └── layout.html
```

Inside of it I run `npx cypress open`.

The cypress app opens propery:

screen shot 2018-12-04 at 10 31 48

And this is what opens inside of the browser cypress opens when I run any of the example tests:

screen shot 2018-12-04 at 10 31 40

Cypress console output:

```
→ npx cypress open
GET /__/ 200 40.683 ms - 352
GET /favicon.ico 302 9.179 ms - 26
GET /__/ 304 8.686 ms - -
HEAD / 500 53.545 ms - -
HEAD / 500 51.452 ms - -
HEAD / 500 50.321 ms - -
```

### Desired behavior:

I expected that it would load the cypress app with the list of tests on the left side and the content of a page open in the test runner to the right.

### Steps to reproduce:

1. Create a directory
2. Inside it add `views/layout.html` with whatever content
3. Inside the root directory run `npx cypress open`
4. Run any of the cypress tests
5. This opens `views/layout.html` for me

### Versions

Issue happens on:

Cypress 3.1.2 and 3.1.3
OS: OSX El Capitan (10.11.6)
Browser: Chrome 70, Canary 73 and Electron 59

I can test it later in the day on my linux machine to confirm if it happens there as well.

Edit: Edited for clarity

### PR DESCRIPTION
**Pull request by flotwig**
Fixes #2891

Not sure that this won't have an adverse impact, also not sure if there's a cleaner way to accomplish this fix

### CODE REVIEW

**Comment by jennifer-shehane** [comment]
Hey @flotwig, we'd like to have a test covering the layout.html bug, so have a failing test without the line of code you added - then the test passing after the code. This may be a bit tricky to write, so let us know if you need any help when you start.

**Comment by flotwig** [comment]
@jennifer-shehane I decided to do the fix a different (cleaner) way, now no filesystem lookups are made at all for our templates.

I added an e2e test that fails appropriately as you requested.

Also, is there an easier way to create the e2e snapshot besides just copy and pasting from terminal and editing out the timestamps and other variables? I had to manually insert a bunch of whitespace to get this to work haha

**Comment by chrisbreiding** [comment]
@flotwig You don't need to manually create the snapshot. You can create it by running the e2e test locally. To run a single e2e test, run the following from `packages/server`:

```
npm run test-e2e -- --spec 3_issue_2891
```

The `3_issue_2891` can be a glob or any subset of the test name to match 1 or more e2e tests to run.

We usually run that and then check the diff to manually verify that the snapshot looks correct.

To update a snapshot, prepend the command with `SNAPSHOT_UPDATE=1`.

**Comment by flotwig** [comment]
Thanks @chrisbreiding, I'll use that info next time

File: packages/server/lib/server.coffee [hunk 1]
```diff
@@ -70,7 +70,14 @@ class Server
 
     ## set the cypress config from the cypress.json file
     app.set("view engine", "html")
-    app.engine("html",     hbs.__express)
+
+    ## since we use absolute paths, configure express-handlebars to not automatically find layouts
+    ## https://github.com/cypress-io/cypress/issues/2891
+    app.engine "html", exphbs {
```
Comment (chrisbreiding) [inline packages/server/lib/server.coffee]: Just a stylistic thing, but we prefer to always use parentheses when calling functions in CoffeeScript. The only exception is tests, where we forgo them for `it`, `describe`, etc.

```coffeescript
app.engine("html", exphbs({
# ...
}))
```

File: packages/server/test/e2e/3_issue_2891_spec.coffee [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+Fixtures = require("../support/helpers/fixtures")
+
+describe "e2e issue 2891", ->
+  e2e.setup()
+
+  ## https://github.com/cypress-io/cypress/issues/2891
+
+  it "passes", ->
+    e2e.exec(@, {
+      project: Fixtures.projectPath('default-layout')
```
Comment (chrisbreiding) [inline packages/server/test/e2e/3_issue_2891_spec.coffee]: Another style thing. Always use double quotes in CoffeeScript for consistency's sake. Some day we'll have all this converted to JS and properly linted 😅

**Comment by brian-mann** [comment]
Whoops, I was going to pair with @flotwig 	on this today to talk about testing strategies, when to write an e2e test vs a unit test... I guess i should have requested changes to indicate not to merge this

**Pull request merged**